composer/trainer/trainer.py

```diff
@@ -543,6 +545,9 @@ class Trainer:
 
             This parameter only controls how many checkpoints are kept locally; checkpoints are not deleted from
             artifact stores.
+        autoresume (bool, optional): Whether or not to enable autoresume, which allows jobs to be killed and restarted
+            to continue training. This parameter requires ``save_folder`` and ``run_name`` to be specified and 
+            ``save_overwrite`` to be ``False``. (default: ``False``)
```

Comment:
For the docstring here, can you elaborate on the logic that is in the notion (e.g. how load_path and load_weights only are internalized), how it finds the checkpoints to restore, and maybe run through a few hypothetical examples (no code required)

Reply:
Updated! Feel free to suggest more changes